src/operator/numpy/linalg/np_norm_forward.cc

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*!
+ * Copyright (c) 2019 by Contributors
+ * \file np_norm_forward.cc
+ * \brief CPU registration of np.linalg.norm
+ */
+
+#include "./np_norm-inl.h"
+
+namespace mxnet {
+namespace op {
+
+NNVM_REGISTER_OP(_npi_norm)
+.describe(R"code()code" ADD_FILELINE)
+.set_num_inputs(1)
+.set_num_outputs(4)
+.set_attr<nnvm::FNumVisibleOutputs>("FNumVisibleOutputs",
+  [](const NodeAttrs& attrs) { return 1; })
+.set_attr_parser(ParamParser<NumpyNormParam>)
+.set_attr<mxnet::FInferShape>("FInferShape", NumpyNormShape)
+.set_attr<nnvm::FInferType>("FInferType", ElemwiseType<1, 4>)
```

Comment:
I think you should probably implement your own `NumpyNormType` function to better match official NumPy's behavior:
```
>>> import numpy as np
>>> a = np.ones((3, 4), dtype=np.int64)
>>> np.linalg.norm(a, ord=2, axis=1).dtype
dtype('float64')
```

Reply:
This could be a bit tricky, because `ReduceAxesComputeImpl` does not support different input output data types. I will have to cast all TBlobs on the fly since I cannot determine the extra space needed.

Reply:
Currently a warning is thrown if the input is integer. There are some issues with reduce APIs and casting TBlobs on the fly is kinda onerous to implement.